lib/AssemblyUtil/TypeToFasta.py

```diff
@@ -0,0 +1,137 @@
+
+    def metagenome_obj_to_fasta(self, ref, obj_type, fasta_dict):
+
+        if 'KBaseMetagenomes.BinnedContigs' in obj_type:
+            fasta_paths = []
+            try:
+                # Binned_contigs_to_file saves fasta file to a directory in scratch.
+                # Path: scratch/binned_contig_files_EXTENSION/Bin#.fasta
+                bin_file_dir = self.mgu.binned_contigs_to_file({'input_ref': ref, 'save_to_shock': 0}) \
+                    ['bin_file_directory']
+                for (dirpath, dirnames, filenames) in os.walk(bin_file_dir):
+                    for fasta_file in filenames:
+                        # For fasta file in the binned contigs directory, copy fasta directly to scratch
+                        # New path: scratch/Bin#.fasta
+                        fasta_path = os.path.join(self.scratch, fasta_file)
+                        copyfile(os.path.join(bin_file_dir, fasta_file), fasta_path)
+                        fasta_paths.append(fasta_path)
+                # Input data into object dict
+                fasta_dict[ref] = {'paths' : fasta_paths, 'type': obj_type}
+
+            # Catch MetagenomeUtil Error
+            except _MGUError as mgue:
+                self.log('Logging exception loading binned contigs to file.')
+                self.log(str(mgue))
+                raise
+
+        return fasta_dict
+
+    def type_to_fasta(self, ref_lst):
+        """type_to_fasta takes in a list of KBase objects references. The object type of each reference
+        is checked in functions: assembly_obj_to_fasta, metagenome_obj_to_fasta, and genome_obj_to_fasta. Depending
+        on the type of KBase object input a fasta file is made through one of the functions mentioned above
+        and a fasta object dictionary is created with structure: {ref: {'path' : fasta_paths, 'type': object type} }
+        for objects of type AssemblySet and GenomeSet a parent ref key-value pair is added such that the structure is:
+        {ref: {'path' : fasta_paths, 'type': object type, 'parent_ref': ref} } """
+
+
+
+        # Initiate objects
+        fasta_dict = {}
+
+        # Get type info for each ref in ref_lst
+        for idx, ref in enumerate(ref_lst):
+
+            # Get KBase object type with get_object_info3
+            obj_info = self.ws.get_object_info3({"objects": [{"ref": ref}]})
+            obj_type = obj_info["infos"][0][2]
+            # Put object in object specific fasta dictionary by type
+            fasta_dict_genome_obj = self.genome_obj_to_fasta(ref, obj_type, fasta_dict)
+            fasta_dict_assembly_obj = self.assembly_obj_to_fasta(ref, obj_type, fasta_dict_genome_obj)
+            fasta_dict_metagenome_obj = self.metagenome_obj_to_fasta(ref, obj_type, fasta_dict_assembly_obj)
+            # Append all individual object dictionaries to complete fasta dictionary for reference list
+            fasta_dict = {**fasta_dict_genome_obj, **fasta_dict_assembly_obj, **fasta_dict_metagenome_obj}
```

Comment:
Not a huge deal, but I think you can either pass `fasta_dict` as an arg to the three functions (`genome_obj_to_fasta`, `assembly_obj_to_fasta`, `metagenome_obj_to_fasta`) or use this line.

Reply:
should be functional either way, but adds a little bit of overhead doing both